x-pack/legacy/plugins/maps/public/actions/map_actions.js

```diff
@@ -78,12 +80,42 @@ async function syncDataForAllLayers(getState, dispatch, dataFilters) {
   const state = getState();
   const layerList = getLayerList(state);
   const syncs = layerList.map(layer => {
-    const loadingFunctions = getLayerLoadingCallbacks(dispatch, layer.getId());
+    const loadingFunctions = getLayerLoadingCallbacks(dispatch, state, layer.getId());
     return layer.syncData({ ...loadingFunctions, dataFilters });
   });
   await Promise.all(syncs);
 }
 
+function cancelRequest(requestToken, state) {
+  if (!requestToken) {
+    return;
+  }
+
+  const cancelCallback = getCancelRequestCallbacks(state).get(requestToken);
+  if (cancelCallback) {
+    cancelCallback();
+    unregisterCancelCallback(requestToken, state);
+  }
+}
+
+function registerCancelCallback(requestToken, callback, state) {
+  getCancelRequestCallbacks(state).set(requestToken, callback);
+}
+
+function unregisterCancelCallback(requestToken, state) {
+  getCancelRequestCallbacks(state).delete(requestToken);
+}
```

Comment:
`registerCancelCallback` and `unregisterCancelCallback` follow an odd pattern in my opinion. I'd expect to find any state updates in the reducer. That's not to say there aren't multiple ways we _could_ go about this, but I do think it's important we're consistent in our approach.